Signal/src/UIView+OWS.h

```diff
@@ -60,6 +60,31 @@ CGFloat ScaleFromIPhone5(CGFloat iPhone5Value);
 
 - (void)centerOnSuperview;
 
+#pragma mark - RTL
+
+// For correct right-to-left layout behavior, use "leading" and "trailing",
+// not "left" and "right".
+//
+// NOTE: the margin values are inverted in RTL layouts.
+- (BOOL)isRTL;
+- (NSArray<NSLayoutConstraint *> *)autoPinLeadingAndTrailingToSuperview;
+- (NSLayoutConstraint *)autoPinLeadingToSuperView;
+- (NSLayoutConstraint *)autoPinLeadingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToSuperView;
+- (NSLayoutConstraint *)autoPinTrailingToSuperViewWithMargin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToTrailingOfView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinLeadingToView:(UIView *)view margin:(CGFloat)margin;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view;
+- (NSLayoutConstraint *)autoPinTrailingToView:(UIView *)view margin:(CGFloat)margin;
+// Return Right on LTR and Right on RTL.
+- (NSTextAlignment)textAlignmentUnnatural;
+// Leading and trailing anchors honor layout margins.
+// When using a UIView as a "div" to structure layout, we don't want it to have margins.
++ (UIView *)containerView;
+- (void)setHLayoutMargins:(CGFloat)value;
```

Comment:
> But the real difference is that our methods use layoutMarginsGuide anchors, which behave differently than the PureLayout alternatives you indicated. Honoring layoutMargins is particularly important in cell layouts, where it lets us align with the complicated built-in behavior of table and collection view cells' default contents.

PureLayout also offers a flavor of functions that honors margins rather than edges. 

e.g.  `- autoPinEdgeToSuperviewMargin:(withInset:)`

How are these new methods different?

My basis for skepticism is that RTL is not some fringe thing and PureLayout is not some fringe thing. I'd be surprised if they didn't have built in the tools for solving these problems - but of course it's possible! Thanks for bearing with me. I'm less familiar with the problem space and want to make sure we don't go all in on a "Weird Signal Way" of doing things if there's already an established convention.